UnityProject/Assets/Scripts/Actors/PlayerController.cs

```diff
@@ -55,11 +59,22 @@ private void Move()
             animator.SetFloat("speed_front",actualSpeed*_inputManager.Vertical);
             animator.SetFloat("speed_right",actualSpeed*_inputManager.Horizontal);
             _movementDirection = new Vector2(_inputManager.Horizontal, _inputManager.Vertical);
-            _velocity = _movementDirection * actualSpeed;
-            rb.velocity = _velocity;
             
-            if (_inputManager.Dash)
+            //If character is moving diagonally, multiply the speed vector by 0.7. Else the character will move at the normal speed * sqrt 2
```

Comment:
This comment is really misleading, you are always multiplying by sqrt 2 (right?) but only in this case do you *also* multiply by .7. That comment makes me think you only ever do one, which would break the intended logic.